lib/screen/book_editor.dart

```diff
@@ -67,6 +74,38 @@ class _BookEditorState extends State<BookEditor> {
       ),
       ),
       ),
+    floatingActionButtonLocation: FloatingActionButtonLocation.endFloat,
+    floatingActionButton: FloatingActionButton.extended(
+    backgroundColor: Theme.of(context).accentColor,
+    onPressed: () => _onSave(uid),
+    heroTag: null,
+    icon: Icon(Icons.check),
+    label: Text("Save"),
+    ),
     );
   }
+
+  Future _onSave(String uid) async {
+    var newFileName = _bookTitleController.text.replaceAll(new RegExp(r"\s+"), "").toLowerCase();
+    final newFile = await changeFileNameOnly(File(_book.cover), newFileName);
+    StorageUploadTask uploadTask = storageReference(newFile.path).putFile(newFile);
+    await uploadTask.onComplete;
+    print('File Uploaded');
+    await storageReference(newFile.path).getDownloadURL().then((fileURL) {
+        _book.cover = fileURL;
+        print(fileURL);
+    });
+    _book.title = _bookTitleController.text;
+    _book.addBook(uid);
+    Navigator.pop(context);
+    Navigator.of(context).pushReplacement(MaterialPageRoute(builder: (context) => HomeScreen()));
+  }
+
+  Future<File> changeFileNameOnly(File file, String newFileName) {
+    var path = file.path;
+    var ext = Path.extension(path);
+    var lastSeparator = path.lastIndexOf(Platform.pathSeparator);
+    var newPath = path.substring(0, lastSeparator + 1) + newFileName + "$ext";
+    return file.rename(newPath);
+  }
```

Comment:
Maybe we should document what this function is doing?

Reply:
Ok got it